### ISSUE TITLE
Error in API call: Resource '<GUID>' does not exist or one of its queried reference-property objects are not present.

### ISSUE THREAD

**Issue by kinnectus** [microsoft/o365-moodle#1735]
We currently run Moodle 3.9.7 with local_o365 plugin v3.9.3 (2020071505).

Since only a couple of weeks ago we've started to have issues where, when we enable Teams creation for a course, the Team is created, however no users are being added to the Team and we can see the following error:

```
Execute scheduled task: Create user groups in Microsoft 365 (local_o365\task\groupcreate)
... started 09:42:27. Current memory use 41.8MB.
Created class team 76f0ce31-9cb5-4978-95f1-a895a2927b93 for course #78307
Recorded group object (76f0ce31-9cb5-4978-95f1-a895a2927b93) into object table with record id 54579
Recorded team object (76f0ce31-9cb5-4978-95f1-a895a2927b93) into object table with record id 54580
Syncing group membership for course #78307
Syncing to group "76f0ce31-9cb5-4978-95f1-a895a2927b93"
Could not sync users to group for course #78307. **Reason: Error in API call: Resource '76f0ce31-9cb5-4978-95f1-a895a2927b93' does not exist or one of its queried reference-property objects are not present.**
Processed courses: 1
... used 17 dbqueries
... used 3.0817511081696 seconds
Scheduled task complete: Create user groups in Microsoft 365 (local_o365\task\groupcreate)
```

Our Moodle courses have users with Team owner (Teacher) and users with Team member (Student) capabilities.

Coincidentally, if we unenrol any of the users from the synced Moodle course and then re-enrol the same users they are successfully added to the Team and they receive the notification within their Teams application and can access the Team as intended.

**Comment by weilai-irl** [comment]
Hi @kinnectus,

I confirm this could be caused by a delay between the time when the team is created, and when the created team is recognised in subsequent API calls to add team owners/members. The delay was small enough before, but it looks like something has changed on the Microsoft side recently which resulted in prolonged delay.

I'll try to add a few retries in the next release to tackle the issue.

Regards,
Lai

**Comment by weilai-irl** [comment]
Hi @kinnectus,

After checking the latest version of the plugin, I realised that this exact issue has been addressed in 3.9.4 release of the plugin.

Please upgrade to the latest version of the plugin (3.9.6 release, version 2020071509), which also contains the fix. I'll close the issue now.

Regards,
Lai
